### ISSUE TITLE
Wrong NS links

### ISSUE THREAD

**Issue by berezovskyi** [OSLC/oslc-site-hugo#265]
Problems identified:

- some folders have stray files (eg Core has `core-vocab` and `core` files)
- some folders should not have been published (eg `/config` has no corresponding PSD)
- some folders have unapproved vocabs in them (eg CM)

https://github.com/oslc-op/website-content-negotiation

https://github.com/oslc-op/website/tree/master/content/ns

---

**TODOs**

- [ ] figure out what is `basicProfile/ldbp` (tracked under https://github.com/oslc-op/website/issues/266)
- [x] keep ./am/architecture-management-vocab.rdf (new) or revert to ./am/am.rdf (removed)
- [x] keep ./asset/asset-management-vocab.rdf or revert to ./asset/asset.rdf
- [ ] _keep ./rm/requirements-management-vocab.rdf or revert to ./rm/rm.rdf (postponed until https://github.com/oslc-op/oslc-specs/issues/240 is resolved)_
- [x] keep ./cm/cm.rdf (unchanged) or replace with ./cm/change-mgt-vocab.rdf (new)
- [x] keep ./core/core-vocab.rdf or ./core/core.rdf
- [x] update Config spec URI
- [x] update RM spec URI
- [x] update QM spec URI
- [x] update AM URI
- [x] https://oslc-op.github.io/oslc-specs/specs/asset/asset-management-vocab.html 
- [x] Auto URI
- [x] PerfMon URI
- [ ] _TRS vocab and URI (postponed until https://github.com/oslc-op/oslc-specs/issues/241 is resolved)_
- [ ] update all the links on https://open-services.net/specifications/vocabularies/ to match
- [x] link to https://open-services.net/specifications/vocabularies/ from the specs page
- [ ] delete the .htaccess file (we are using nginx)

**Comment by berezovskyi** [comment]
- Updated all XML files except for TRS (Turtle for Config).
- Regenerated old Turtle/JSON-LD files with Jena 3.12
- Generated N-Triples for all vocabs

**Comment by ndjc** [comment]
The vocabulary links from open-services.net/ns are still wrong. They now redirect to the specs, or to a page that in turn contains links only to the specifications, and not to the vocabularies.

For example, if I go to open-services.net/ns/rm, I get redirected to https://archive.open-services.net/bin/view/Main/RmSpecificationV2a8d9.html. That is not the vocabulary - indeed, it doesn't even have a link to the vocabulary. To find that, I have to manually click on the RM Home button on the breadcrumbs at the top of the page, from there follow the link to the new wiki, and then click on the link to the RM 2.0 vocab at https://archive.open-services.net/bin/view/Main/RmVocabulary.html#.

Similarly, if I just go to open-services.net/ns, I get a page of spec listings, with links that claim to be to the vocabularies, but in fact point to the shape tables in the specs (this is really a separate issue - the vocab links on that page https://open-services.net/specifications/vocabularies/ are all incorrect).

**Comment by berezovskyi** [comment]
My fault. And I was so proud of my effort to update all the links! 🤦‍♂️

I will update RM when I get home. If got any other vocabulary links, please share them. I don’t think I saw them immediately when I was searching for the right links to the main parts of the specs.

--
/Andrew
(from phone)

30 sep. 2019 kl. 17:04 skrev Nick Crossley <notifications@github.com<mailto:notifications@github.com>>:


The vocabulary links from open-services.net/ns<http://open-services.net/ns> are still wrong. They now redirect to the specs, or to a page that in turn contains links only to the specifications, and not to the vocabularies.

For example, if I go to open-services.net/ns/rm<http://open-services.net/ns/rm>, I get redirected to https://archive.open-services.net/bin/view/Main/RmSpecificationV2a8d9.html. That is not the vocabulary - indeed, it doesn't even have a link to the vocabulary. To find that, I have to manually click on the RM Home button on the breadcrumbs at the top of the page, from there follow the link to the new wiki, and then click on the link to the RM 2.0 vocab at https://archive.open-services.net/bin/view/Main/RmVocabulary.html#.

Similarly, if I just go to open-services.net/ns<http://open-services.net/ns>, I get a page of spec listings, with links that claim to be to the vocabularies, but in fact point to the shape tables in the specs (this is really a separate issue - the vocab links on that page https://open-services.net/specifications/vocabularies/ are all incorrect).

**Comment by berezovskyi** [comment]
Will fix tomorrow at it requires redeploy of the content neg app

**Comment by berezovskyi** [comment]
@jamsden @ndjc please go over https://open-services.net/specifications/vocabularies/ and give me a set of actions to do.

**Comment by ndjc** [comment]
I've completed the initial verification of the RDF vocabulary pages at open-services.net, and I've found quite a few inconsistencies or errors that need to be addressed. Some will need discussion. See the attached note.
[OSLC V2 Vocabulary Status.docx](https://github.com/oslc-op/website/files/4195025/OSLC.V2.Vocabulary.Status.docx)

**Comment by berezovskyi** [comment]
@ndjc in https://github.com/oslc-op/oslc-admin/blob/master/minutes/2021/2021-03-18.md#ns-deploy agreed to publish vocabs+shapes for all specs as soon as they hit PS stage. This renders old vocab recovery efforts obsolete for: Core, RM, CM, QM, and Query. Please advice on actions needed to recover the right files and redirects for the other specs that are still published incorrectly.

**Comment by ndjc** [comment]
Once we rule out Core, RM, CM, QM, and Query, the only remaining two issues are minor (an outdated RDF vocabulary source for AM, and a broken link for Asset Management) are both minor; I'm OK with closing this entire issue as won't fix.

**Comment by berezovskyi** [comment]
>  a broken link for Asset Management

do you have the right one? 

> an outdated RDF vocabulary source for AM

OK, we can update it when it goes to PS. @jamsden no objections?

**Comment by jamsden** [comment]
none

**Comment by ndjc** [comment]
re the asset management issue, my note says:

> https://archive.open-services.net/wiki/asset-management/OSLC-Asset-Management-Vocabulary/index.html# is correct, but the link in that page to the RDF version of the file is broken - it just refers back to itself, the HTML page.

so if I go to open-services.net/ns/asset#, it redirects **correctly** to https://archive.open-services.net/wiki/asset-management/OSLC-Asset-Management-Vocabulary/index.html#. That page then states:

> Following [W3C best practices](http://www.w3.org/TR/swbp-vocab-pub/), this information is available in HTML format (this page) and in [RDFS format](https://archive.open-services.net/wiki/asset-management/OSLC-Asset-Management-Vocabulary/index.html#).

The second link is a bad one - it points back to the same page. Later in the page, there is a more correct reference to https://archive.open-services.net/ns/asset/asset.rdf, which works correctly.

In other words, this is a content error in the original open-services.net wiki content, and is nothing to do with current open-services redirects. Since this issue was originally about correcting references to existing old content, and since we're not currently working on asset management, I see no urgency to fix this defect in the original content.

**Comment by berezovskyi** [comment]
I have updated the vocabs in #287 but did not update the link in the https://github.com/oslc-op/website-content-negotiation/blob/master/negotiation.js#L45. Please check if we want to update it but then it's not really an errata but instead lowering the bar even further down to a PSD.

**Comment by berezovskyi** [comment]
@ndjc thanks for catching that. I am fairly certain we don't want the link to point to https://archive.open-services.net/ns/asset/asset.rdf but https://open-services.net/ns/am/am.rdf (does not work now because it's not handled by nginx but the conneg server but we can fix it if needed) or replace the asset.rdf with a new asset.ttl or the newly generated asset.rdf

Wait a second! I though we were updating the AM vocab, not Asset!

**Comment by berezovskyi** [comment]
The vocabs have been updated, shapes were added and vocab/shape page links were updated in https://github.com/oslc-op/website-content-negotiation/pull/10, https://github.com/oslc-op/website/pull/292, https://github.com/oslc-op/website/pull/290, https://github.com/oslc-op/website/pull/293, https://github.com/oslc-op/website/pull/289

curl tests passed for all content-types:

```
Running production tests
Running tests on 'ns/core'
Running tests on 'ns/core/shapes/3.0'
Running tests on 'ns/cm'
Running tests on 'ns/cm/shapes/3.0'
Running tests on 'ns/rm'
Running tests on 'ns/rm/shapes/2.1'
Running tests on 'ns/qm'
Running tests on 'ns/qm/shapes/2.1'
Running tests on 'ns/config'
Running tests on 'ns/am'
Running tests on 'ns/asset'
Running tests on 'ns/auto'
Running tests on 'ns/perfmon'
ALL TESTS PASSED
```
